tests/ssl-opt.sh

```diff
@@ -9009,8 +9009,8 @@ run_test    "TLS 1.3: HelloRetryRequest check - gnutls" \
             "$P_CLI debug_level=4 force_version=tls13" \
             1 \
             -c "received HelloRetryRequest message" \
-            -c "HRR not supported" \
-            -c "Last error was: -0x6E00 - SSL - The handshake negotiation failed" \
+            -c "tls13 client state: MBEDTLS_SSL_CLIENT_HELLO(1)" \
```

Comment:
We should check that the parsing of the HRR was successful.

Reply:
If you add "--disable-client-cert" to the GnuTLS server parameters the handshake succeeds as in the OpenSSL case. Good to see that the 2nd client hello is already almost done.